### PR TITLE
Improve Job Summary

### PR DESCRIPTION
**Pull request by attiasas**
- [x] The pull request is targeting the `dev` branch.
- [x] The code has been validated to compile successfully by running `go vet ./...`.
- [x] The code has been formatted properly using `go fmt ./...`.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-security/actions/workflows/analysis.yml) passed.
- [x] All [tests](https://github.com/jfrog/jfrog-cli-security/actions/workflows/test.yml) have passed. If this feature is not already covered by the tests, new tests have been added.
- [x] All changes are detailed at the description. if not already covered at [JFrog Documentation](https://github.com/jfrog/documentation), new documentation have been added.

-----

Depends on: 
* https://github.com/jfrog/jfrog-cli-core/pull/1230
* https://github.com/jfrog/jfrog-cli-core/pull/1245

Adjust to changes in the Job Summary Interface
We will now create:
1. Security section (Old curation table, minor UI changes)
2. Provide dynamic scan summary markdown for results base on the given data files. (recording with index and args as unique keys for ecosystem infrastructure)

New Generated Security Summary Section:
image

New Generated results:
image
image
image
image

### CODE REVIEW

File: commands/scan/buildscan.go [hunk 1]
```diff
@@ -183,7 +183,13 @@ func (bsc *BuildScanCommand) runBuildScanAndPrintResults(xrayManager *xray.XrayS
 			}
 		}
 	}
-	err = utils.RecordSecurityCommandOutput(utils.ScanCommandSummaryResult{Results: scanResults.GetSummary(), Section: utils.Build})
+	err = utils.RecordSecurityCommandSummary(utils.NewBuildScanSummary(
+		scanResults,
+		bsc.serverDetails,
+		bsc.includeVulnerabilities,
+		bsc.buildConfiguration.GetProject() != "",
```
Comment (hadarshjfrog) [inline commands/scan/buildscan.go:190]: couldn't we get violations from watches? we assume only Project in this case...

File: utils/results.go [hunk 2]
```diff
@@ -81,41 +81,12 @@ func (r *Results) IsIssuesFound() bool {
...
-	return
-}
-
-// Returns a summary for the provided targets. If no targets are provided, a summary for all targets is returned.
-func (r *Results) getScanSummaryByTargets(targets ...string) (summaries []formats.ScanSummaryResult) {
-	if len(targets) == 0 {
-		// No filter, one scan summary for all targets
-		summaries = append(summaries, getScanSummary(r.ExtendedScanResults, r.ScaResults...))
-		return
+func (r *Results) CountScanResultsFindings(includeVulnerabilities, includeViolations bool) (total int) {
```
Comment (hadarshjfrog) [inline utils/results.go:84]: I see we always pass (true,true) to this function - and from what I see - we never return the vuln - only violations

File: commands/scan/buildscan.go (hunk 1 above)
Comment (attiasas) [reply commands/scan/buildscan.go:190]: no watches flag for build scan, only project can be provided as violation context in this cmd
image

File: utils/results.go (hunk 2 above)
Comment (attiasas) [reply utils/results.go:84]: For the first note - I can remove the inputs to the func.
About the second note - we return the violations count if exists (> 0) or the vulnerabilities count if no violations
image

Comment (hadarshjfrog) [reply utils/results.go:84]: ok - so if there are violations - we'll never show the vulnerabilities

File: go.mod [hunk 3]
```diff
@@ -114,7 +114,8 @@ require (
 	gopkg.in/warnings.v0 v0.1.2 // indirect
 )
 
-// replace github.com/jfrog/jfrog-cli-core/v2 => github.com/jfrog/jfrog-cli-core/v2 dev
+// EyalDelarea:new_summary_ui
```
Comment (hadarshjfrog) [inline go.mod:117]: stating the obvious - but don't forget to revert

File: utils/severityutils/severity.go [hunk 4]
```diff
@@ -30,6 +33,45 @@ const (
...
+	case Medium:
+		return "🟠"
+	case Low:
+		return "🟡"
+	default:
+		return "⚪️"
+	}
+}
+
+func getSeveritySvgIcon(severity Severity) string {
```
Comment (hadarshjfrog) [inline utils/severityutils/severity.go:58]: I don't see we ever use this - traced back and the only call to it gets "false"...

File: utils/resultwriter.go [hunk 5]
```diff
@@ -606,3 +607,179 @@ func IsEmptyScanResponse(results []services.ScanResponse) bool {
...
+	}
+	return
+}
+
+func getScanViolationsSummary(extendedScanResults *ExtendedScanResults, scaResults ...*ScaScanResult) (violations *formats.ScanViolationsSummary) {
+	watches := datastructures.MakeSet[string]()
+	failBuild := false
+	scanIds := []string{}
+	moreInfoUrls := []string{}
+	vioUniqueFindings := map[ViolationIssueType]formats.ResultSummary{}
```
Comment (hadarshjfrog) [inline utils/resultwriter.go:662]: please use violationsUniqueFindings

File: utils/resultwriter_test.go [hunk 6]
```diff
@@ -441,3 +442,154 @@ func TestJSONMarshall(t *testing.T) {
...
+		includeVulnerabilities bool
+		includeViolations      bool
+		expected               formats.ResultsSummary
+	}{
+		{
+			name:                   "Vulnerabilities only",
+			includeVulnerabilities: true,
+			results: &Results{
+				ScaResults: []*ScaScanResult{{
+					Target:      "target1",
```
Comment (hadarshjfrog) [inline utils/resultwriter_test.go:508]: try to set Target value with a more meaningful name

File: utils/securityJobSummary.go [hunk 7]
```diff
@@ -1,553 +1,600 @@
...
-	"github.com/owenrumney/go-sarif/v2/sarif"
-	"golang.org/x/exp/maps"
 )
 
 const (
-	Build    SecuritySummarySection = "Builds"
-	Binary   SecuritySummarySection = "Artifacts"
-	Modules  SecuritySummarySection = "Modules"
-	Curation SecuritySummarySection = "Curation"
+	Build    SecuritySummarySection = "Build-info Scans"
```
Comment (hadarshjfrog) [inline utils/securityJobSummary.go:23]: if `Modules` representes `"Source Code Scans"` - we should probably rename it to something more "source code" related - it's not clear that modules refer only to code.

File: utils/securityJobSummary.go [hunk 8]
```diff
@@ -1,553 +1,600 @@
...
+	Binary   SecuritySummarySection = "Artifact Scans"
+	Modules  SecuritySummarySection = "Source Code Scans"
+	Docker   SecuritySummarySection = "Docker Image Scans"
+	Curation SecuritySummarySection = "Curation Audit"
+
+	PreFormat     HtmlTag = "<pre>%s</pre>"
+	ImgTag        HtmlTag = "<img alt=\"%s\" src=%s>"
+	CenterContent HtmlTag = "<div style=\"display: flex; align-items: center; text-align: center\">%s</div>"
+	BoldTxt       HtmlTag = "<b>%s</b>"
+	Link          HtmlTag = "<a href=\"%s\">%s</a>"
```
Comment (hadarshjfrog) [inline utils/securityJobSummary.go:33]: I'd suggest renaming the vars that has more than one %s in them - e.g, SummaryAndDetails.
In Link or ImgTag I'm still not sure myself - but when we have completely two texts - we should be clearer

File: utils/securityJobSummary_test.go [hunk 9]
```diff
@@ -6,184 +6,279 @@ import (
...
 
+	"github.com/jfrog/jfrog-cli-core/v2/artifactory/utils/commandsummary"
 	"github.com/jfrog/jfrog-cli-security/formats"
+	"github.com/jfrog/jfrog-cli-security/utils/jasutils"
 	"github.com/stretchr/testify/assert"
 )
 
 var (
 	summaryExpectedContentDir = filepath.Join("..", "tests", "testdata", "other", "jobSummary")
+
```
Comment (hadarshjfrog) [inline utils/securityJobSummary_test.go:17]: even it's tests - let's use valid url. suggest adding `.jfrog.io` / `.jfrog.com`

File: utils/securityJobSummary_test.go [hunk 10]
```diff
@@ -6,184 +6,279 @@ import (
...
+					Vulnerabilities: &formats.ScanResultSummary{ScaResults: &formats.ScaScanResultSummary{ScanIds: []string{TestScaScanId}, MoreInfoUrls: []string{"https://test-url"}}},
+				}},
+			}},
+		},
+		{
+			name:                "Violations - Not defined",
+			index:               commandsummary.BinariesScan,
+			violations:          true,
+			expectedContentPath: filepath.Join(summaryExpectedContentDir, "violations_not_defined.md"),
+			args:                &ResultSummaryArgs{BaseJfrogUrl: testPlatformUrl},
```
Comment (hadarshjfrog) [inline utils/securityJobSummary_test.go:111]: same comment on urls - let's try the test data to be a bit more realistic

File: formats/summary_test.go [hunk 11]
```diff
@@ -1,199 +1,413 @@
...
+			assert.Equal(t, testCase.expectedApproved, testCase.curatedPackages.GetApprovedCount())
+			assert.Equal(t, testCase.expectedBlocked, testCase.curatedPackages.GetBlockedCount())
 		})
 	}
 }
 
-func TestTwoLevelSummaryCount(t *testing.T) {
+func TestResultSummary(t *testing.T) {
+	testSummary := ResultSummary{
+		"High":    map[string]int{NoStatus: 1, "Status1": 2},
```
Comment (hadarshjfrog) [inline formats/summary_test.go:50]: Please be consistent with the quotes

File: formats/summary_test.go [hunk 12]
```diff
@@ -1,199 +1,413 @@
...
+		{"Empty", CuratedPackages{}, 0, 0},
+		{"Approved", CuratedPackages{PackageCount: 1}, 1, 0},
+		{"Blocked", CuratedPackages{Blocked: []BlockedPackages{{Packages: map[string]int{"npm://test:1.0.0": 1}}}, PackageCount: 1}, 0, 1},
+		{
+			"Multiple",
+			CuratedPackages{
+				Blocked: []BlockedPackages{
+					{
+						Policy:    "Test",
+						Condition: "Test condition",
```
Comment (hadarshjfrog) [inline formats/summary_test.go:26]: can we have more descriptive texts here?
I saw some examples of "aged packages", "immature", "highProfile", same note in condition "cvss > 7.5"

File: utils/severityutils/severity.go (hunk 4 above)
Comment (attiasas) [reply utils/severityutils/severity.go]: removed

File: utils/resultwriter.go (hunk 5 above)
Comment (attiasas) [reply utils/resultwriter.go]: done

File: utils/resultwriter_test.go [hunk 13]
```diff
@@ -441,3 +442,154 @@ func TestJSONMarshall(t *testing.T) {
...
+			}),
+		},
+	}
+
+	expectedVulnerabilities := &formats.ScanResultSummary{
+		ScaResults: &formats.ScaScanResultSummary{
+			ScanIds:      []string{TestScaScanId},
+			MoreInfoUrls: []string{TestMoreInfoUrl},
+			Security: formats.ResultSummary{
+				"Critical": map[string]int{jasutils.ApplicabilityUndetermined.String(): 1},
```
[the remark this answers is not in the document]
Comment (attiasas) [reply utils/resultwriter_test.go:473]: yeah but the status is different so its they are different issues

File: utils/resultwriter_test.go (hunk 6 above)
Comment (attiasas) [reply utils/resultwriter_test.go:508]: most of those tests will be changed in the refactor

File: utils/securityJobSummary.go (hunk 7 above)
Comment (attiasas) [reply utils/securityJobSummary.go:23]: Source code are grouped under a module/Application. this value is not used now also this is just an enum

File: utils/securityJobSummary.go (hunk 8 above)
Comment (attiasas) [reply utils/securityJobSummary.go:33]: done (you can also hover over it and see the actual string)

File: utils/securityJobSummary_test.go (hunk 9 above)
Comment (attiasas) [reply utils/securityJobSummary_test.go:17]: done

File: utils/securityJobSummary_test.go (hunk 10 above)
Comment (attiasas) [reply utils/securityJobSummary_test.go:111]: its little as possible but also try to cover most cases in code

File: formats/summary_test.go (hunk 11 above)
Comment (attiasas) [reply formats/summary_test.go:50]: ?

File: formats/summary_test.go (hunk 12 above)
Comment (attiasas) [reply formats/summary_test.go:26]: you have realistic in `securityJobSummary_test` where you can actually view the UI that is generated.
This test is only for logic of counting

File: formats/summary_test.go (hunk 11 above)
Comment (hadarshjfrog) [reply formats/summary_test.go:50]: Add quotes to "NoStatus" - not only to "Status1"

**Review by hadarshjfrog — APPROVED** [review]

**Pull request merged**